src/components/modals/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import { ButtonWithModal } from './ButtonWithModal';
+import { ActionButton } from './ModalWithActionButtons';
+
+export interface ButtonConfirmationModalProps {
+  modalTitle: string;
+  modalContentText: string;
+  name: string;
+  confirmButtonProps: ActionButton;
+  cancelButtonProps: ActionButton;
+}
+
+export const ButtonWithConfirmationModal = ({
+  modalTitle,
+  modalContentText,
+  name,
+  confirmButtonProps,
+  cancelButtonProps,
+  ...otherOpenButtonProps
+}: ButtonConfirmationModalProps) => {
+  return (
+    <ButtonWithModal
+      modalTitle={modalTitle}
+      modalContentText={modalContentText}
+      name={name}
+      actionButtonList={[cancelButtonProps, confirmButtonProps]}
```

Comment:
i'd like to see default text for confirmButton and cancelButton. (or the very least cancel)
if you're cancelling then you're just gonna cancel